nrpy/helpers/safewrite.py

```diff
@@ -0,0 +1,87 @@
+"""
+The Safewrite class helps users see and understand changes to generated output files.
+
+Author: Steven R. Brandt
+"""
+
+from typing import Any, Optional, Union
+import io
+import os
+from difflib import context_diff
+import sys
+from subprocess import Popen, PIPE
+from pathlib import Path
+from nrpy.helpers.colored import colored
+
+try:
+    from clang_format import _get_executable as get_executable  # type: ignore[import-untyped]
+except ModuleNotFoundError:
+
+    def get_executable(_: str) -> str:
+        """Use /bin/cat if clang_format is not available."""
+        return "/bin/cat"
+
+
+clang_formatter = get_executable("clang-format")
+
+verbose = False
+nochange = False
+
+
+class SafeWrite:
+    """
+    SafeWrite ensures that files are only written if their contents would change.
+    In addition, it prints a message to the screen notifying the user if the file has changed.
+    If verbose is set, it will print a diff of the change.
+    If nochange is set, then it will avoid changing file contents at all. This setting
+    is useful when debugging.
+    """
+
+    # black insists on indenting this way, pylint does not allow it
```

Comment:
Please run pylint from the NRPy root directory. It has a compatible .pylintrc. Remove this comment if there's no issue.

Reply:
Doiing that does not fix the issue